frontend/lib/parsing/chpl.ypp

```diff
@@ -3135,6 +3121,10 @@ for_expr_base:
   {
     $$ = context->buildGeneralLoopExpr(@$, @2, @4, $1, nullptr, $2, $4);
   }
+| for_loop_kw zippered_iterator TDO expr %prec TFOR
+  {
+    $$ = context->syntax(@$, "expression level loops with 'zip' must have an index.");
```

Comment:
Unless precedent suggests otherwise, I think "expression-level" should be hyphenated

Reply:
Updated here and for `expression-level coforall loops are not supported`